editor/src/components/canvas/canvas-strategies/strategies/group-conversion-helpers.ts

```diff
@@ -925,24 +973,50 @@ export function createPinChangeCommandsForElementBecomingGroupChild(
   if (
     elementMetadata == null ||
     isLeft(elementMetadata.element) ||
-    !isJSXElement(elementMetadata.element.value)
+    !elementCanBeAGroupChild(elementMetadata.element.value)
   ) {
     throw new Error(
       `Invariant violation: ElementInstanceMetadata.element found for ${EP.toString(
         expectedPath,
-      )} was null or Left or not JSXElement`,
+      )} cannot be a group child`,
     )
   }
+
   const childLocalRect: LocalRectangle = canvasRectangleToLocalRectangle(
     forceFiniteRectangle(elementMetadata.globalFrame),
     globalBoundingBoxOfAllElementsToBeWrapped,
   )
-  return commandsForPinChangeCommandForElementBecomingGroup(
-    expectedPath,
-    elementMetadata.element.value.props,
-    childLocalRect,
-    newLocalRectangleForGroup,
-  )
+  switch (elementMetadata.element.value.type) {
+    case 'JSX_ELEMENT':
+      return commandsForPinChangeCommandForElementBecomingGroup(
+        expectedPath,
+        elementMetadata.element.value.props,
+        childLocalRect,
+        newLocalRectangleForGroup,
+      )
+    case 'JSX_CONDITIONAL_EXPRESSION':
+      const activeBranch = getConditionalActiveCase(
+        elementMetadata.elementPath,
+        elementMetadata.element.value,
+        jsxMetadata,
+      )
+      const branch =
+        activeBranch === 'true-case'
+          ? elementMetadata.element.value.whenTrue
+          : elementMetadata.element.value.whenFalse
+      return isJSXElement(branch)
+        ? commandsForPinChangeCommandForElementBecomingGroup(
+            expectedPath,
+            branch.props,
+            childLocalRect,
+            newLocalRectangleForGroup,
+          )
+        : []
```

Comment:
I think we should not arrive to this case because we should check earlier what is the active branch of the conditional (see my comment above)